jbpm-bpmn2/src/main/java/org/jbpm/bpmn2/xml/SubProcessHandler.java

```diff
@@ -80,6 +81,8 @@ public Object end(final String uri, final String localName,
 			    Boolean isAsync = Boolean.parseBoolean((String)node.getMetaData().get("customAsync"));
 				// create new timerNode
 				ForEachNode forEachNode = new ForEachNode();
+                // indirect to the real node if it is an async event node
+                forEachNode.getCompositeNode().getMetaData().put(NodeInstance.METADATA_WRAP_ASYNC_NODE, true);
```

Comment:
I struggle with this explanation - if it is an asynce event node, mark wrap async node true - but in the end when you call useAsync method, it will do the negation, which means it won't wrap it. Moreover, the condition to wrap/set this property only if it is an async event node is quite misleading since it will be always done for every multiInstanceLoopCharacteristics node - just by reading the code.